Lib/test/test_syntax.py

```diff
@@ -2607,6 +2607,14 @@ def test_continuation_bad_indentation(self):
 
         self.assertRaises(IndentationError, exec, code)
 
+    @support.cpython_only
+    def test_disallowed_type_param_names(self):
+        # See gh-128632
```

Comment:
Could you also test the following related cases:

- Using a generic function or type alias instead of a class with these names for the type parameters
- Using the names `__classcell__` and `__classdictcell__`, which appear internally in the implementation.

Reply:
Added func and alias tests. Didn't add `__classcell__` or `__classdictcell__` tests as those are not excluded and don't cause problems currently. Do you want to add them to the forbidden list?
```py
>>> class A:
...     class B[__classcell__]: print(type(__classcell__))
...     
<class 'typing.TypeVar'>
>>> class A:
...     class B[__classdictcell__]: print(type(__classdictcell__))
...     
<class 'typing.TypeVar'>
```

Reply:
Thanks for adding the tests! I want the `*cell*` names to also be tested, because they're also somewhat likely to cause problems, even if they don't currently crash.

Reply:
So I interpret 'somewhat likely to cause problems' to mean yes disallow these names (and test them).

Reply:
Since they work fine now we should not disallow them, especially because we'd want to backport this change. However, we should have a test to make sure future changes don't introduce crashes with these names.

Reply:
Names re-allowed and tests added. Also re-allowed `__class__` as that doesn't crash either even though it is special.